py/openage/convert/dataformat/exportable.py

```diff
@@ -404,11 +405,61 @@ def structs(cls):
         return ret
 
     @classmethod
+    def format_hash(cls, hasher=None):
+        """
+        provides a deterministic hash of all exported structure members
+
+        used for determining changes in the exported data, which requires
+        data reconversion.
+        """
+
+        if not hasher:
+            hasher = hashlib.sha512()
+
+        # struct properties
+        hasher.update(cls.name_struct.encode())
+        hasher.update(cls.name_struct_file.encode())
+        hasher.update(cls.struct_description.encode())
+
+        # only hash exported struct members!
+        # non-exported values don't influence anything.
+        members = cls.get_data_format(
+            allowed_modes=(True, READ_EXPORT, NOREAD_EXPORT),
+            flatten_includes=False,
+        )
+        for is_parent, export, member_name, member_type in members:
```

Comment:
this for-loop iterates over the members, so i thought sticking them together might point that out.